backend/pkg/database/odatasql/jsonsql/postgres.go

```diff
@@ -37,7 +37,7 @@ func (postgres) CastToDateTime(strTime string) string {
 }
 
 func (postgres) JSONEach(source string) string {
-	return fmt.Sprintf("JSONB_ARRAY_ELEMENTS((%s))", source)
+	return fmt.Sprintf("JSONB_ARRAY_ELEMENTS(CASE JSONB_TYPEOF(%s) WHEN 'array' THEN (%s) ELSE '[]' END)", source, source)
```

Comment:
From the PR: 
Ensures that null values are handled correctly by JSONB_EACH_ARRAY. The postgres function expect the data must be an array, so we need to detect any other types in the SQL statement and switch it to empty string.

Maybe worth adding this comment? is it "to empty array"?